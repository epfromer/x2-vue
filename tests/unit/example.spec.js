import { createLocalVue, shallowMount } from "@vue/test-utils";
import HelloWorld from "@/components/HelloWorld.vue";
import VueRouter from "vue-router";
import Vuetify from "vuetify";
import Vuex from "vuex";

describe("HelloWorld.vue", () => {
  const localVue = createLocalVue();
  localVue.use(VueRouter);
  localVue.use(Vuex);
  localVue.use(Vuetify);

  const vuetify = new Vuetify({
    icons: {
      iconfont: "mdi"
    }
  });

  const router = new VueRouter({
    routes: []
  });

  const store = new Vuex.Store({
    state: {}
  });

  function doMount(mountType) {
    return mountType(HelloWorld, {
      localVue,
      vuetify,
      store,
      router
    });
  }

  it("isVueInstance", () => {
    const tree = doMount(shallowMount);
    expect(tree.isVueInstance()).toBeTruthy();
  });

  it("toMatchSnapshot", () => {
    const tree = doMount(shallowMount);
    expect(tree.element).toMatchSnapshot();
  });

  // it("renders props.msg when passed", () => {
  //   const msg = "new message";
  //   const wrapper = shallowMount(HelloWorld, {
  //     propsData: { msg }
  //   });
  //   expect(wrapper.text()).toMatch(msg);
  // });
});
