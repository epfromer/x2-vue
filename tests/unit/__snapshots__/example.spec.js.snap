// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HelloWorld.vue toMatchSnapshot 1`] = `
<v-container-stub
  tag="div"
>
  <v-layout-stub
    tag="div"
    text-center=""
    wrap=""
  >
    <v-flex-stub
      tag="div"
      xs12=""
    >
      <v-img-stub
        class="my-3"
        contain="true"
        height="200"
        options="[object Object]"
        position="center center"
        src=""
        transition="fade-transition"
      />
    </v-flex-stub>
     
    <v-flex-stub
      mb-4=""
      tag="div"
    >
      <h1
        class="display-2 font-weight-bold mb-3"
      >
        
        Welcome to Vuetify
      
      </h1>
       
      <p
        class="subheading font-weight-regular"
      >
        
        For help and collaboration with other Vuetify developers,
        
        <br />
        please join our online
        
        <a
          href="https://community.vuetifyjs.com"
          target="_blank"
        >
          Discord Community
        </a>
      </p>
    </v-flex-stub>
     
    <v-flex-stub
      mb-5=""
      tag="div"
      xs12=""
    >
      <h2
        class="headline font-weight-bold mb-3"
      >
        What's next?
      </h2>
       
      <v-layout-stub
        justify-center=""
        tag="div"
      >
        <a
          class="subheading mx-3"
          href="https://vuetifyjs.com/components/api-explorer"
          target="_blank"
        >
          
          Explore components
        
        </a>
        <a
          class="subheading mx-3"
          href="https://vuetifyjs.com/layout/pre-defined"
          target="_blank"
        >
          
          Select a layout
        
        </a>
        <a
          class="subheading mx-3"
          href="https://vuetifyjs.com/getting-started/frequently-asked-questions"
          target="_blank"
        >
          
          Frequently Asked Questions
        
        </a>
      </v-layout-stub>
    </v-flex-stub>
     
    <v-flex-stub
      mb-5=""
      tag="div"
      xs12=""
    >
      <h2
        class="headline font-weight-bold mb-3"
      >
        Important Links
      </h2>
       
      <v-layout-stub
        justify-center=""
        tag="div"
      >
        <a
          class="subheading mx-3"
          href="https://vuetifyjs.com"
          target="_blank"
        >
          
          Documentation
        
        </a>
        <a
          class="subheading mx-3"
          href="https://community.vuetifyjs.com"
          target="_blank"
        >
          
          Chat
        
        </a>
        <a
          class="subheading mx-3"
          href="https://madewithvuejs.com/vuetify"
          target="_blank"
        >
          
          Made with Vuetify
        
        </a>
        <a
          class="subheading mx-3"
          href="https://twitter.com/vuetifyjs"
          target="_blank"
        >
          
          Twitter
        
        </a>
        <a
          class="subheading mx-3"
          href="https://medium.com/vuetify"
          target="_blank"
        >
          
          Articles
        
        </a>
      </v-layout-stub>
    </v-flex-stub>
     
    <v-flex-stub
      mb-5=""
      tag="div"
      xs12=""
    >
      <h2
        class="headline font-weight-bold mb-3"
      >
        Ecosystem
      </h2>
       
      <v-layout-stub
        justify-center=""
        tag="div"
      >
        <a
          class="subheading mx-3"
          href="https://github.com/vuetifyjs/vuetify-loader"
          target="_blank"
        >
          
          vuetify-loader
        
        </a>
        <a
          class="subheading mx-3"
          href="https://github.com/vuetifyjs/vuetify"
          target="_blank"
        >
          
          github
        
        </a>
        <a
          class="subheading mx-3"
          href="https://github.com/vuetifyjs/awesome-vuetify"
          target="_blank"
        >
          
          awesome-vuetify
        
        </a>
      </v-layout-stub>
    </v-flex-stub>
  </v-layout-stub>
</v-container-stub>
`;
